.gitlab/choco_build/choco_build.yml

```diff
@@ -23,23 +23,30 @@ windows_choco_offline_7_x64:
     paths:
       - omnibus/pkg
 
-# The online version of the choco job fetches the msi package from S3 so
-# it is run only once the msi package is pushed
+# The online version of the choco job gets the msi package through the gitlab artifacts
 windows_choco_online_7_x64:
   rules:
-    !reference [.on_deploy_stable_or_beta_repo_branch_manual]
+    !reference [.on_deploy_stable_or_beta_repo_branch]
   stage: choco_and_install_script_build
   tags: ["runner:windows-docker", "windowsversion:1809"]
-  needs: ["deploy_staging_windows_tags-7"]
+  needs: ["deploy_staging_windows_tags-7", "deploy_packages_windows-x64-7"]
   variables:
     ARCH: "x64"
   script:
     - '$_instance_id = (iwr  -UseBasicParsing http://169.254.169.254/latest/meta-data/instance-id).content ; Write-Host "Running on instance $($_instance_id)"'
     - $ErrorActionPreference = "Stop"
-    - if (Test-Path .omnibus) { remove-item -recurse -force .omnibus }
+    - mkdir temp\
+    - copy omnibus\pkg\*.msi temp\
+    - if (Test-Path omnibus) { remove-item -recurse -force omnibus }
```

Comment:
Why do we need to empty this folder? (I'm assuming that's what this does)

Reply:
Yes, it does empty the folder. The `omnibus/pkg` directory is passed to the `choco_deploy` job ([artifact](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/664814427/artifacts/browse/omnibus/pkg/)). It is removed and created again to ensure that it starts at a clean slate and doesn't send any unnecessary files to the `choco_deploy` job. The `choco_deploy` job is still in the datadog-agent pipeline to ensure there is a backup way to trigger chocolatey releases.

Reply:
It might be safer (and overall simpler) to put the artifacts in a different location instead.

I wasn't sure how strong I should suggest this within the context of this PR but then I realized that clearing up this folder and then setting it as an artifact could have unexpected effects that we may prefer to protect ourselves against: if other job further down the pipeline was to require an artifact from the same folder that was created during a previous stage, I _think_ it would no longer be able to find it.

Reply:
Yes, I think that putting the artifacts in another location would be much simpler. However, I'm keeping the omnibus/pkg folder to maintain compatibility with the existing `choco_deploy` job in **datadog-agent**. To automate chocolatey I'm moving `choco_deploy` to **agent-release-management** and the artifacts would be retrieved from the internal S3 bucket instead through gitlab artifacts. Technically the existing `choco_deploy` job in **datadog-agent** can be removed but I'm keeping it as a backup way of manually publishing choco packages.

I don't think it makes sense to invest time in changing the directory structure now, since it would require modifying the `choco_deploy` job in **datadog-agent**, which we're aiming to phase out. I will definitely add a task to ensure that once the chocolatey task has been automated succesfully to remove any unnecessary jobs.

Reply:
Yes, I think that putting the artifacts in another location would be much simpler. However, I'm keeping the omnibus/pkg folder to maintain compatibility with the existing choco_deploy job in datadog-agent. To automate chocolatey I'm moving choco_deploy to agent-release-management and the artifacts would be retrieved from the internal S3 bucket instead through gitlab artifacts. Technically the existing choco_deploy job in datadog-agent can be removed but I'm keeping it as a backup way of manually publishing choco packages.

I don't think it makes sense to invest time in changing the directory structure now, since it would require modifying the choco_deploy job in datadog-agent, which we're aiming to phase out. I will definitely create a ticket to ensure that once the chocolatey task has been automated succesfully to remove any unnecessary jobs.

Reply:
Wouldn't this just involve a small change to these lines (just updating the `omnibus\pkg` references)?
https://github.com/DataDog/datadog-agent/blob/0257eaa323d19c3a5354707cc7a70d1aa60c91df/.gitlab/choco_deploy/choco_deploy.yml#L21-L24

But it's true that it might not that big of a deal given that we can remove all of it once `choco_deploy` is no longer in this pipeline, so I think I'm fine with this and leave the decision up to you.